modules/cosmosBidAdapter.js

```diff
@@ -0,0 +1,398 @@
+    if (impression) {
+      impressions.push(impression);
+    }
+    return impressions;
+  },
+
+  /**
+   * create impression (single) object
+   **/
+  _creatImpression: function (request) {
+    if (!request.hasOwnProperty(MEDIA_TYPES)) {
+      return undefined;
+    }
+
+    var params = request && request.params ? request.params : null;
+    var impression = {
+      id: request.bidId ? request.bidId : spec._createUUID(),
+      secure: window.location.protocol === HTTPS ? 1 : 0,
+      bidfloorcur: request.params.currency ? request.params.currency : DEFAULT_CURRENCY
+    };
+    if (params.bidFloor) {
+      impression.bidfloor = params.bidFloor;
+    }
+
+    if (params.tagId) {
+      impression.tagid = params.tagId.toString();
+    }
+
+    var banner;
+    var video;
+    var mediaType;
+    for (mediaType in request.mediaTypes) {
+      switch (mediaType) {
+        case BANNER:
+          banner = spec._createBanner(request);
+          if (banner) {
+            impression.banner = banner;
+          }
+          break;
+        case VIDEO:
+          video = spec._createVideo(request);
+          if (video) {
+            impression.video = video;
+          }
+          break;
+      }
+    }
+
+    return impression.hasOwnProperty(BANNER) ||
+      impression.hasOwnProperty(VIDEO) ? impression : undefined;
+  },
+
+  /**
+   * create the banner object
+   **/
+  _createBanner: function (request) {
+    if (utils.deepAccess(request, 'mediaTypes.banner')) {
+      var banner = {};
+      var sizes = request.mediaTypes.banner.sizes;
+      if (sizes && utils.isArray(sizes) && sizes.length > 0) {
+        var format = [];
+        banner.w = sizes[0][0];
+        banner.h = sizes[0][1];
+        sizes.forEach(size => {
+          format.push({
+            w: size[0],
+            h: size[1]
+          });
+        });
+        banner.format = format;
+      }
+
+      spec._copyObject(request.mediaTypes.banner, banner);
+      spec._copyObject(request.params.banner, banner);
+      return banner;
+    }
+    return undefined;
+  },
+
+  /**
+   * create video object
+   **/
+  _createVideo: function (request) {
+    if (utils.deepAccess(request, 'mediaTypes.video')) {
+      var video = {};
+      var sizes = request.mediaTypes.video.playerSize;
+      if (sizes && utils.isArray(sizes) && sizes.length > 1) {
+        video.w = sizes[0];
+        video.h = sizes[1];
+      }
+      spec._copyObject(request.mediaTypes.video, video);
+      spec._copyObject(request.params.video, video);
+      return video;
+    }
+    return undefined;
+  },
+
+  /**
+   * create site object
+   **/
+  _createSite: function (request) {
+    var rSite = request.params.site;
+    if (rSite || !request.params.app) {
+      var site = {};
+      spec._copyObject(rSite, site);
+      site.domain = utils.getTopWindowLocation().hostname;
+      site.ref = utils.getTopWindowReferrer();
+      site.page = utils.getTopWindowLocation().href;
```

Comment:
These three `getTopWindow...` functions are marked to be deprecated and may not be available in the next major release.  A `refererInfo` object is available in the `bidderRequest` param that should have values available you can use instead.  

The following link provides additional information:
http://prebid.org/dev-docs/bidder-adaptor.html#referrers

Can you please update these values accordingly?

Reply:
Thanks for input and saving our future efforts. Updated above code with refererInfo object.